test/inductor/test_torchinductor.py

```diff
@@ -243,6 +244,43 @@ def run(*ex, **kwargs):
                 if exact_dtype:
                     assert correct_val.dtype == actual_val.dtype
 
+    if check_gradient:
+        def gather_leaf_tensors(args, kwargs):
+            leaf_tensors = []
+            args, args_spec = tree_flatten(args)
+            kwargs, kwargs_spec = tree_flatten(kwargs)
+            args = args + kwargs
+            for arg in args:
+                if not isinstance(arg, torch.Tensor):
+                    continue
+                if arg.requires_grad:
+                    leaf_tensors.append(arg)
+            return leaf_tensors
+
+        def compute_grads(args, kwrags, results):
+            flat_results, _ = tree_flatten(results)
+            flat_diff_results = [r for r in flat_results if r.requires_grad]
+            assert len(flat_diff_results) > 0
+
+            grads = [torch.ones(r.shape, device=r.device, dtype=r.dtype) for r in flat_diff_results]
```

Comment:
better to have random grads, `torch.ones` can hide the bugs

Reply:
This might be numerically unstable for big Tensor, for gradcheck, we use a unit-norm Tensor as it proved to be the most stable.

Reply:
Updated to use unit norm now.